BluetoothLeChat/app/src/main/java/com/example/bluetoothlechat/bluetooth/ChatServer.kt

```diff
@@ -0,0 +1,287 @@
+package com.example.bluetoothlechat.bluetooth
+
+import android.app.Application
+import android.bluetooth.*
+import android.bluetooth.le.AdvertiseCallback
+import android.bluetooth.le.AdvertiseData
+import android.bluetooth.le.AdvertiseSettings
+import android.bluetooth.le.BluetoothLeAdvertiser
+import android.content.Context
+import android.os.ParcelUuid
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.bluetoothlechat.bluetooth.Message.RemoteMessage
+import com.example.bluetoothlechat.chat.DeviceConnectionState
+
+private const val TAG = "ChatServer"
+
+object ChatServer {
+
+    // hold reference to app context to run the chat server
+    private var app: Application? = null
+    private lateinit var bluetoothManager: BluetoothManager
+    // BluetoothAdapter should never be null since BLE is required per
+    // the <uses-feature> tag in the AndroidManifest.xml
+    private val adapter: BluetoothAdapter = BluetoothAdapter.getDefaultAdapter()
```

Comment:
Would it make sense to add a comment about how this could be `null` if launched from an emulator that doesn't have bluetooth? I tried launching the app on an emulator and got a crash on launch.

Reply:
That definitely makes sense, I'll add this information and update the existing content to better explain things